src/java/org/apache/cassandra/repair/RepairRunnable.java

```diff
@@ -267,16 +257,19 @@ public void run()
 
     private void runMayThrow() throws Exception
     {
-        ActiveRepairService.instance.recordRepairStatus(cmd, ParentRepairStatus.IN_PROGRESS, ImmutableList.of());
+        state.phase.setup();
+        ActiveRepairService.instance.recordRepairStatus(state.cmd, ParentRepairStatus.IN_PROGRESS, ImmutableList.of());
 
         List<ColumnFamilyStore> columnFamilies = getColumnFamilies();
         String[] cfnames = columnFamilies.stream().map(cfs -> cfs.name).toArray(String[]::new);
 
         this.traceState = maybeCreateTraceState(columnFamilies);
-
         notifyStarting();
-
         NeighborsAndRanges neighborsAndRanges = getNeighborsAndRanges();
+        // We test to validate the start JMX notification is seen before we compute neighbors and ranges
+        // but in state watching, we rely on getNeighborsAndRanges to know where we are running repair...
+        // TODO fix this mismatch (fix test?)
```

Comment:
is it completed?

Reply:
nope, and didn't think it was too big a deal to keep this way so removed the TODO